src/duke/core/Duke.java

```diff
@@ -57,6 +65,37 @@ public void run() {
 
 
             switch(keyword[0]) {
+                case FIND:
+                    ArrayList<Task> result = searchTask(task, keyword[1]);
+                    try {
+                        for (int i = 0; i < result.size(); i++) {
+                            String s = indentation;
+                            String s2 = "";
+
+                            if (result.get(i) instanceof Todo) {
+                                s += (i + 1) + "." + " [T]";
+                                s2 = result.get(i).getName();
+                            } else if (result.get(i) instanceof Deadline) {
+                                s += (i + 1) + "." + " [D]";
+                                s2 = result.get(i).getName() +  " ( " + ((Deadline) result.get(i)).getTime() + " )";
+                            } else if (result.get(i) instanceof Event){
+                                s += (i + 1) + "." + " [E]";
+                                s2 = result.get(i).getName() +  " ( " + ((Event) result.get(i)).getTime() + " )";
+                            }
+                            if (result.get(i).isDone() == false) {
+                                s += "[ ] " + s2;
+                                System.out.println(s);
+                            } else {
+                                s += "[X] " + s2;
+                                System.out.println(s);
+                            }
```

Comment:
Your code is too deeply-nested, with many layers of conditional branches. and the variable naming are a bit non-comprehensible. Do consider breaking down your code and make variables more obvious for readers to understand what it is about